bin/cron/build_contact_rollups

```diff
@@ -0,0 +1,428 @@
+  log "Updating professional learning enrollment"
+  PEGASUS_DB.run "
+    UPDATE contact_rollups,
+      (SELECT user_id, GROUP_CONCAT(course) AS courses FROM
+         (SELECT DISTINCT pd_enrollments.user_id, pd_workshops.course FROM #{DASHBOARD_DB_NAME}.pd_enrollments AS pd_enrollments
+            INNER JOIN #{DASHBOARD_DB_NAME}.pd_workshops AS pd_workshops ON pd_workshops.id = pd_enrollments.pd_workshop_id
+            WHERE course IN (#{COURSE_LIST})
+            ORDER BY 1,2
+         ) q
+      GROUP by user_id
+      ) src
+    SET contact_rollups.professional_learning_enrolled = src.courses
+    WHERE contact_rollups.dashboard_user_id = src.user_id"
+  log "   Completed in " + ((Time.now) - start).to_s + " seconds"
+end
+
+def update_professional_learning_attendance
+  start = Time.now
+  log "Updating professional learning attendance"
+  PEGASUS_DB.run "
+    UPDATE contact_rollups,
+      (SELECT teacher_id, GROUP_CONCAT(course) as courses FROM
+        (SELECT DISTINCT teacher_id, course FROM
+          (SELECT pd_attendances.teacher_id, pd_workshops.course
+            FROM #{DASHBOARD_DB_NAME}.pd_attendances AS pd_attendances
+              INNER JOIN #{DASHBOARD_DB_NAME}.pd_sessions AS pd_sessions ON pd_sessions.id = pd_attendances.pd_session_id
+              INNER JOIN #{DASHBOARD_DB_NAME}.pd_workshops AS pd_workshops ON pd_workshops.id = pd_sessions.pd_workshop_id
+            WHERE course IN (#{COURSE_LIST})
+            UNION
+            SELECT teacher_id,
+              CASE program_type
+                WHEN 1 THEN 'CS in Science'
+                WHEN 2 THEN 'CS in Algebra'
+                WHEN 3 THEN 'Exploring Computer Science'
+                WHEN 4 THEN 'CS Principles'
+                WHEN 5 THEN 'CS Fundamentals'
+              END AS course
+            FROM #{DASHBOARD_DB_NAME}.workshop_attendance AS workshop_attendance
+              INNER JOIN #{DASHBOARD_DB_NAME}.segments AS segments ON segments.id = workshop_attendance.segment_id
+              INNER JOIN #{DASHBOARD_DB_NAME}.workshops AS workshops ON workshops.id = segments.workshop_id
+            ) q1
+          ORDER BY 1,2
+          ) q2
+        GROUP BY teacher_id
+      ) src
+    SET contact_rollups.professional_learning_attended = src.courses
+    WHERE contact_rollups.dashboard_user_id = src.teacher_id"
+  log "   Completed in " + ((Time.now) - start).to_s + " seconds"
+end
+
+def mysql_multi_connection
+  # return a connection with the MULTI_STATEMENTS flag set that allows multiple statements in one DB call
+  Sequel.connect(CDO.pegasus_db_writer.sub('mysql:', 'mysql2:'), flags: ::Mysql2::Client::MULTI_STATEMENTS)
+end
+
+def update_geo_data_from_forms(form_kind)
+  start = Time.now
+  log "Updating geo data for form kind #{form_kind}"
+
+  record_count = 0
+  num_updated = 0
+  num_parse_errors = 0
+  update_batch = ""
+  conn = mysql_multi_connection
+  time_last_output = start
+
+  PEGASUS_DB[:forms].where(kind: form_kind).each do |form|
```

Comment:
Do we want to batch this at all so not all forms are loaded into memory at once?